src/ray/protobuf/gcs_service.proto

```diff
@@ -767,18 +772,59 @@ message AddTaskEventDataReply {
 message GetTaskEventsRequest {
   // Filter object where predicates are AND together.
   message Filters {
-    // Get task events from a job.
-    optional bytes job_id = 1;
-    // Get task events from a set of tasks.
-    repeated bytes task_ids = 2;
-    // Get the task events with an actor id.
-    optional bytes actor_id = 3;
-    // Get the task events of task with names.
-    optional string name = 4;
+
+    // The filter to filter on the job id of a task.
+    message JobIdFilter {
+      // The prediate of the filter.
+      FilterPredicate predicate = 1;
+      // The job id to filter on.
+      bytes job_id = 2;
+    }
+
+    // The filter to filter on the task id.
+    message TaskIdFilter {
+      // The prediate of the filter.
+      FilterPredicate predicate = 1;
+      // The task id to filter on.
+      bytes task_id = 2;
+    }
+
+    // The filter to filter on the actor id of a task.   
+    message ActorIdFilter {
+      // The prediate of the filter.
+      FilterPredicate predicate = 1;
+      // The actor id to filter on.
+      bytes actor_id = 2;
+    }
+
+    // The filter to filer on the task name.
+    message NameFilter {
+      // The predicate of the filter.
+      FilterPredicate predicate = 1;
+      // The task name to filter on.
+      string name = 2;
+    }
+
+    // The filter to filter on the latest state of a task.
+    message StateFilter {
+      // The predicate of the filter.
+      FilterPredicate predicate = 1;
+      // The latest state of the task to filter on.
+      string state = 2;
+    }
+
+    // Task events filter by a job id.
+    optional JobIdFilter job_filter = 1;
+    // Task events filter by a set of task filters.
```

Comment:
Will add the comment.

Agree that `== TASK1 OR != TASK2` would be weird. At the same time, today, we don't have a constraint on how the filter can be set and it is possible for the user to set a filter like that. I'll need to also add the constraints and make it clear in the docs to avoid confusions.

Reply:
Actually, I have some more findings after testing with the list ask state API. 

From what I see, although in the GCS processing logic, the task_ids are OR together, in the `transform` logic in the `list_tasks` function: 
https://github.com/ray-project/ray/blob/b3b6f2cc5989b012026c0046fdbd28f6b8e3a85a/python/ray/dashboard/state_aggregator.py#L461
All the filters will be processed again with `AND` sematics. 

This means that, with today's logic, it seems that the task ids are actually AND together. And from a code search, I don't see any logic is using the GCS `HandleGetTaskEvents`. If that's the case, I think we should fix the internal GCS API to be aligned with the existing publish API logic using `AND` semantics. 

@rynewang Not sure if I missed any historical context and wondering what's your take on it?

Reply:
Synced offline with @jjyao. We should make the GCS behavior be consistent with the behavior in the API server which is to make all task_ids AND together. Will update the code.

Reply:
I think sticking with `AND` is fine given we don't really have a way to specify `OR` from the frontend yet without some complex query parsing. 

I guess if we do with `AND`, what's the usecase there's repeated id filters? Guess if we don't do `OR` here, we just need a single task id filter?

Reply:
I think the `AND` semantics should still make sense in the "!=" scenario when the same task id is specified multiple times and the user want to filter out multiple task ids. 

At the same time, it makes me think whether we want to make other filters also have multiple filters so that more filters can be pushed down to GCS.